.github/workflows/rust.yml

```diff
@@ -81,6 +81,45 @@ jobs:
       - name: Run clippy
         run: cargo clippy --no-deps -- -Dclippy::all -Wclippy::pedantic
 
+  external-deps:
+    if: github.event_name == 'push' || github.event_name == 'schedule' ||
+      github.event.pull_request.head.repo.full_name != github.repository
+
+    name: Test with external dependencies
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly
+
+      - name: Cache Cargo registry
+        uses: Swatinem/rust-cache@v2
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@v2
+        with:
+          tool: nextest
+
+      - name: Build
+        run: cargo build
+
+      - name: Run the external dependencies
+        run: docker compose up -d
+
+      - name: Test
+        run: cargo nextest run --all-features -- --include-ignored
+
+      - name: Test docs
+        run: cargo test --doc
+
+      - name: Stop the external dependencies
+        run: docker compose down
```

Comment:
I wonder how much it is actually needed. When the testing is done, GitHub will just kill the runner, and this won't even run if the tests don't pass.

Reply:
Hmm, fair enough, I guess there is not much of a point in doing that. Let's remove this.